systems/plants/RigidBodyManipulator.cpp

```diff
@@ -3239,14 +3300,82 @@ int RigidBodyManipulator::findLinkId(string linkname, int robot)
   if(num_match != 1)
   {
     cerr<<"couldn't find unique link "<<linkname<<endl;
-    return(EXIT_FAILURE);
+    return(nullptr);
   }
   else
   {
-    return ind_match;
+    return this->bodies[ind_match];
   }
 }
 
+int RigidBodyManipulator::findLinkId(string name, int robot)
+{
+	shared_ptr<RigidBody> link = findLink(name,robot);
+	if (link == nullptr) return(EXIT_FAILURE);
```

Comment:
I don't think EXIT_FAILURE is appropriate here.  It's value is system dependent and is supposed to be passed into the exit() function.